app/src/main/java/model/place/Place.kt

```diff
@@ -26,34 +26,37 @@ import com.guerinet.mymartlet.util.firestore
  * @author Julien Guerinet
  * @since 1.0.0
  */
-class Place {
-
-    var id: Int = 0
-
-    var name: String = ""
-
-    var categories: List<Int> = listOf()
-
-    var address: String = ""
+data class Place(
+    val id: Int,
+    val name: String,
+    val categories: List<Int>,
+    val address: String,
+    private val courseName: String,
+    val coordinates: GeoPoint
+) {
 
     /** Name of the place when listed under a course location, an empty String if equivalent to the [name]  */
-    var courseName: String = ""
+    val coursePlaceName: String
         get() {
             // If there is no override, simply use the name
-            return if (field.isEmpty()) name else field
+            return if (courseName.isEmpty()) name else courseName
```

Comment:
If `courseName` is only used here, you don't have to keep it and you can just make

`val coursePlaceName = courseName.takeIf { it.isNotEmpty() } ?: name` or something

Reply:
I simplified the getter for the `coursePlaceName` to what you have above. The reason I have to keep `courseName` around is because `Place` is a data class, and so everything within its constructor needs to be a property. The alternative is to not make it a data class. This wouldn't pose a problem now, but might pose a problem down the line should we ever use the stuff that a data class provides. What do you think?

Reply:
If you're comparing courses or using strings, you can keep it as a data class. It isn't a huge deal. Otherwise there's no real need

Reply:
In that case let's leave it as is.